virtcontainers/clh.go

```diff
@@ -391,8 +392,42 @@ func (clh *cloudHypervisor) resizeMemory(reqMemMB uint32, memoryBlockSizeMB uint
 }
 
 func (clh *cloudHypervisor) resizeVCPUs(reqVCPUs uint32) (currentVCPUs uint32, newVCPUs uint32, err error) {
-	clh.Logger().WithField("function", "resizeVCPUs").Warn("not supported")
-	return 0, 0, nil
+	cl := clh.client()
+
+	// Retrieve the number of current vCPUs via HTTP API
+	ctx, cancel := context.WithTimeout(context.Background(), clhAPITimeout*time.Second)
+	info, _, err := cl.VmInfoGet(ctx)
+	if err != nil {
+		clh.Logger().WithField("function", "resizeVCPUs").WithError(openAPIClientError(err)).Info("[clh] VmInfoGet failed")
+		return 0, 0, openAPIClientError(err)
+	}
+	cancel()
```

Comment:
I think you should move this after `ctx, cancel := context.WithTimeout(context.Background(), clhAPITimeout*time.Second)` as `defer cancel()`

Reply:
@devimc What I want to achieve with this `cancel()` here is to reset the timer after the first HTTP API call (`VmInfoGet`), and use a new timer and `cancel()` call for the second HTTP API (`VmResizePut`). I was assuming with a `defer cancel()` here, the timer will be imposed for the  rest of the `resizeVCPUs()` function (not the first HTTP API call `VmResizePut`). 

Basically I want two separate timer for two HTTP API calls. Please let me know if my understanding is incorrect or there is better/nicer way to do it. Thanks a lot.

Reply:
ok, but please add a comment  :smile: